src/app/components/sign-up/sign-up.component.ts

```diff
@@ -18,6 +18,7 @@ export class SignUpComponent {
     const password: string = this.signUpForm.get('password')?.value ?? ''
     if(this.signUpForm.valid){
```

Comment:
This will be valid even if they leave them almost blank. You defaulting to empty strings and then submitting to firebase is not going to be valid inputs for firebase.